### PR TITLE
Dbaas 5392 - Support for basic auth and JWT tokens in mlflow REST api

### PR DESCRIPTION
**Pull request by Ben-Epstein**
This enables users to set their username/password or a JWT token and all future mlflow requests will include the provided authentication information. 

## Description
Now that we have the mlflow 1.15 upgrade, we can include auth headers in requests. This code change enables us to grab the header and perform authorization and authentication. No changes in the cloud are necessary because we call the `__try_auto_login ` function automatically in the cloud, and that will set the necessary variables for.

## Motivation and Context
SSO

## Dependencies
None

## How Has This Been Tested?
See screenshots

## Screenshots (if appropriate):
Using jwt:
image
image

Using basic auth:
image
image

## Changelog Inclusions

- base feature note
  - **BREAKING** note on base feature
  - Basically whatever formatting we have here, just plain-text
	%> command example
- next feature
  - note on next feature

### Additions
Support for SSO compatible authorization and authentication with basic auth and JWT tokens for mlflow API.

### Breaking Changes
You must now set the parameters in `login_director` by name, as they are no longer positional arguments.

### CODE REVIEW

**Review by juanvisoler — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 setup(
     name="splicemachine",
-    version="2.7.0",
+    version="2.7.0dev0",
```
Comment (myles-novick) [inline setup.py:26]: is this on purpose

**Review by abaveja313 — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (Ben-Epstein) [reply setup.py:26]: yup - in line with the new stuff we're doing for ml-workflow. Since master is really things that are "in development" we don't want to accidentally squash the official package versions. So I made it dev0
